src/tracks/SampleTrack.cpp

```diff
@@ -440,8 +448,15 @@ void SampleTCOView::mouseReleaseEvent(QMouseEvent *_me)
 void SampleTCOView::mouseDoubleClickEvent( QMouseEvent * )
 {
 	QString af = m_tco->m_sampleBuffer->openAudioFile();
-	if( af != "" && af != m_tco->m_sampleBuffer->audioFile() )
-	{
+
+	if ( af.isEmpty() ) {} //Don't do anything if no file is loaded
+	else if ( af == m_tco->m_sampleBuffer->audioFile() )
+	{	//Instead of reloading the existing file, just reset the size
```

Comment:
If a user modifies one of their audio files (not entirely unlikely; could be new vocals, updated bounced track, new export from another DAW, etc.), opening the file dialog and re-selecting the sample will no longer load the modified file. IMO this is unintuitive.

Reply:
I didn't introduce this behavior. Existing version skips reloading entirely, my patch makes it skip reloading but at least reset size

Should i remove the check entirely?

Reply:
Ah, indeed. I didn't read the original carefully enough, just saw the comment added and assumed it was new. In that case, I think the condition ought to go, but I don't know whether it's in scope here. Sorry.

Reply:
I'd rather leave as is then since this targets stable.